### PR TITLE
[false-negative] Fix 8947 for consider-using-min/max-builtin

### LINKED ISSUE #8947: False negatives for consider-using-min-builtin/consider-using-max-builtin
### Bug description

`consider-using-min-builtin/consider-using-max-builtin` did not catch next situations:

```
value = 5
max_value = 10

if value > max_value:
    max_value = value

if value > float(max_value):
    value = float(max_value)
```

### Configuration

_No response_

### Command used

```shell
pylint --load-plugins=pylint.extensions.code_style --enable=all test.py
```

### Pylint output

```shell
************* Module pylint_test
pylint_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
pylint_test.py:1:0: C0103: Constant name "value" doesn't conform to UPPER_CASE naming style (invalid-name)
pylint_test.py:2:0: C0103: Constant name "max_value" doesn't conform to UPPER_CASE naming style (invalid-name)
pylint_test.py:5:4: C0103: Constant name "max_value" doesn't conform to UPPER_CASE naming style (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 10.00/10, -6.67)
```

### Expected behavior

First `if` statement should be suggested to be replaced with `max_value = max(max_value, value)`.

Second - with `value = min(value, float(max_value))`.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (main, Jun 26 2023, 16:57:25) [GCC 7.5.0]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by rhyn0**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Altered checks to account for statements written in the reverse order of an expected if block.

Closes #8947

### CODE REVIEW

**Comment by rhyn0** [comment]
@Pierre-Sassoulas When you have a chance, could you kindly review? False Negative fix for `consider-using-min-builtin` / `consider-using-max-builtin`

File: tests/functional/c/consider/consider_using_min_max_builtin.py
Comment (Pierre-Sassoulas) [inline tests/functional/c/consider/consider_using_min_max_builtin.py:83]:
```suggestion

if value > float(value3):
    value = float(value3)
    
offset = 1
if offset + value < min_value:
    min_value = offset + value
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
The primer looks pretty good, there's no false positive there. Do you think it's possible to include the examples from the original issue with a cast to float in order to deal with``offset + value `` on top of the ``Name`` / ``Const`` ? If it's a lot of work we might merge as is and create another issue but the design / implementation would probably benefit from being able to handle this generically, what do you think ?

**Comment by rhyn0** [comment]
I think adding in checks for more complex blocks that use a `BinOp` or `Call` is doable in this PR. I'll probably have to break the `name` `value` getter and comparator into helper functions and then do something about it there. Just worrying about the case of handling something like below

```python3
def i_return_float_2(*args):
  return 2.0

value = 1
value3 = 2
if value < i_return_float_2(value3, True, False, 3, 5, 9):
  value = i_return_float_2(value3, True, False, 3, 5, 9)
```

Which I guess becomes trivial once I know that those nodes are `Call` and just compare the list of args. 

I'll take a crack at it and try

**Comment by Pierre-Sassoulas** [comment]
I guess a hacky way to make it work would be to call the ``as_string`` function on the node to check if the same representation is used, but I don't know the perf implication of doing that (could be good actually, considering everything that need to be taken into account if we handle each node type differentely).

**Comment by rhyn0** [comment]
Actually kind of like this idea. Might be able to escape the 'isinstance' ladder that is duplicated throughout that function.

**Comment by rhyn0** [comment]
Added in the original false negatives, and some additional true negatives to the test cases.

The addition of increasing scope of `consider-using-max-builtin` caused some pylint errors to be raised in other test cases i.e. `access_attr_before_def_false_positive.py` so added another ignore code for that.

Took @Pierre-Sassoulas idea of using the `node.as_string()` method to create a common name for complex operands (`float(value3)`) and then changed the function to only compare string representation instead of sometimes the `value` for the cases of `Const`.

**Comment by Pierre-Sassoulas** [comment]
You probably need to install the test requirements with ``pip install -r requirements_test.txt`` / ``pip install -r requirements_test_pre_commit.txt``

File: tests/functional/a/access/access_attr_before_def_false_positive.py
Comment (Pierre-Sassoulas) [inline tests/functional/a/access/access_attr_before_def_false_positive.py:36]:
```suggestion
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
This look pretty good already, waiting for CI to be green and for another maintainer review before merging but this is definitely going into 3.1 !

**Comment by rhyn0** [comment]
> You probably need to install the test requirements with pip install -r requirements_test.txt / pip install -r requirements_test_pre_commit.txt

Thanks doing this was able to resolve those issues for me locally, let's hope that passes in CI.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
We're going to release 3.1.0 this week-end let's merge 😄

**Comment by Pierre-Sassoulas** [comment]
Congratulation on becoming a pylint contributor 🎉

**Pull request merged**
